### ISSUE TITLE
Don't recommend uncrustify any more

### ISSUE THREAD

**Issue by aljelly** [spheras/desktopfolder#155]
Because:
  - It's not difficult to incorrectly invoke the tool or give incorrect parameters to the tool
  - Different versions may not be compatible with the config, and maintaining the config would take time
  - Different people using different versions may cause the formatting to keep changing each time
  - Automatic code formatters (that aren't included with the language build tools like e.g. Rust, Go) seem to have a reputation of being problematic
  - Considering the above points it's also yet another thing someone has to set up in order to contribute

The first 4 points above can lead to huge amounts of diff noise and/or unpredictable and unwanted changes to the formatting.

If a particular style of code is wanted then it's probably much easier to ask the submitter to format it manually (use the reviews) or do it yourself after. Additionally one could make sure the code is already formatted nicely so that others are more likely to format it in the same way.

Basically [this page](https://github.com/spheras/desktopfolder/wiki/Uncrustifying) would be removed and [this page](https://github.com/spheras/desktopfolder/wiki/Contributing) would be edited to remove the mentioning of Uncrustify.

**Comment by spheras** [comment]
mmm I see. I think it is important to have a common code style, maybe the solution is just to format it later as you comment.  I come from the Java world, and there is everything much easier.. How are solved these problems in really big projects? Do they use the uncrustify tool? Aren't there standards code style for Vala? Doesn't exist a 'plugin' to invoke during the meson/ninja compilation?

**Comment by aljelly** [comment]
I know at least elementary manually reviews the style of PRs and elementary have their own Vala code style that they push. I'm not aware of anything for Vala that comes by default (again unlike Rust or Go where a formatter comes with the compiler tools, but they're widely used and accepted and have a good reputation).

The main motivation for not recommending uncrustify was because in that PR it messed up a lot of the formatting. You wouldn't have to worry about people using the automatic tool and accidentally messing up the formatting up due to the reasons I listed in the original post. And it's more cumbersome to install and set up (probably more than asking people to manually make the styling changes) if it doesn't come with the compiler tools.

**Comment by spheras** [comment]
Yes, I agree. Just wondering how big projects deal with this problem. You comment that elementary review manually the code, OMG! Ok, I think your solution is good: to remove those pages from the wiki, and I can uncrustify manually the code later. Maybe we could remove only the [contributing page ](https://github.com/spheras/desktopfolder/wiki/Contributing) and maintain [the other one](https://github.com/spheras/desktopfolder/wiki/Uncrustifying)... I say this, because for me the uncrustifying wiki is very useful. My intention is to follow the same uncrustify method for the project, but just not to force to contributors to follow the code style (I will uncrustify it later).

Maybe @Jacob-Vlijm or @fossfreedom can hint us how they manage this problem from a big project perspective. Do you let contributors to add code without a style code? Do you review the code style? Do you use uncrustify tool for vala? If yes, do you launch uncrustify automatically on commits? What is your recommendation for this? thanks in advance.

**Comment by aljelly** [comment]
I've made the changes but kept the Uncrustify page (with a small note added at the top) as you requested. The changes can be reverted later if you want to restore them.
